Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -17062,6 +17062,68 @@ void SpeculativeJIT::compileStringLocaleCompare(Node* node)
     strictInt32Result(resultGPR, node);
 }
 
+void SpeculativeJIT::compileStringIndexOf(Node* node)
+{
+    if (node->child3()) {
+        SpeculateCellOperand base(this, node->child1());
+        SpeculateCellOperand argument(this, node->child2());
+        SpeculateInt32Operand index(this, node->child2());
+
+        GPRReg baseGPR = base.gpr();
+        GPRReg argumentGPR = argument.gpr();
+        GPRReg indexGPR = index.gpr();
+
+        speculateString(node->child1(), baseGPR);
+        speculateString(node->child2(), argumentGPR);
+
+        std::optional<UChar> character;
+        String searchString = node->child2()->tryGetString(m_graph);
+        if (!!searchString) {
+            if (searchString.length() == 1)
+                character = searchString.characterAt(0);
+        }
```

Comment:
See below.  This (except for the `SpeculateInt32Operand index(this, node->child2());`) can be moved above the `if (node->child3())` check.

Reply:
Ditto. SpeculateInt32Operand etc. is very sideeffectful thing. We should not do it, and keep each of sideeffectful sequence carefully constructed.